src/annotator/notebook.js

```diff
@@ -29,26 +31,39 @@ export default class Notebook extends Delegator {
     super(element, config);
     this.frame = null;
 
+    this.groupId = null;
+    this.prevGroupId = null;
+
     this.container = document.createElement('div');
     this.container.style.display = 'none';
     this.container.className = 'notebook-outer';
 
-    this.subscribe('showNotebook', () => this.show());
+    this.subscribe('showNotebook', groupId => {
+      this.groupId = groupId;
+      this.show();
+    });
     this.subscribe('hideNotebook', () => this.hide());
     // If the sidebar has opened, get out of the way
     this.subscribe('sidebarOpened', () => this.hide());
   }
 
-  init() {
-    if (!this.frame) {
-      this.frame = createNotebookFrame(this.options);
+  update() {
```

Comment:
Is the `update` method intended to be private?

Reply:
I'll rename it.